### PR TITLE
SDIT-404: 🔧 Initial stab at prisoner opensearch dev instance

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/00-namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: hmpps-prisoner-search-dev
+  labels:
+    cloud-platform.justice.gov.uk/is-production: "false"
+    cloud-platform.justice.gov.uk/environment-name: "dev"
+  annotations:
+    cloud-platform.justice.gov.uk/business-unit: "HMPPS"
+    cloud-platform.justice.gov.uk/application: "hmpps-prisoner-search"
+    cloud-platform.justice.gov.uk/owner: "Digital Prison Services: dps-hmpps@digital.justice.gov.uk"
+    cloud-platform.justice.gov.uk/source-code: "https://github.com/ministryofjustice/hmpps-prisoner-search.git"
```
Comment (petergphillips) [inline namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/00-namespace.yaml:12]: will create the repo once the namespace is up

File: namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/opensearch.tf [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  engine_version = "OpenSearch_2.5"
+
+  cluster_config = {
+    instance_count = 2
+    instance_type  = "t3.small.search"
+  }
+
+  ebs_options = {
+    volume_size = 10
```
Comment (petergphillips) [inline namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/opensearch.tf:21]: no idea what to put here.  We've got an existing elasticsearch instance in `prisoner-offender-search-dev` namespace that should be exactly the same size as this instance will be.

File: namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/opensearch.tf [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  }
+}
+
+module "os_snapshots_s3_bucket" {
```
Comment (petergphillips) [inline namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/opensearch.tf:25]: presumably snapshots are supported for opensearch too 🤷.  Config copied from our existing elasticsearch instance.

File: namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/rds.tf [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  application               = var.application
+  is-production             = var.is_production
+  namespace                 = var.namespace
+  environment-name          = var.environment-name
+  infrastructure-support    = var.infrastructure_support
+  db_instance_class         = "db.t4g.micro"
+  db_engine                 = "postgres"
+  db_engine_version         = "15"
+  rds_family                = "postgres15"
+  deletion_protection       = true
```
Comment (petergphillips) [inline namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/rds.tf:16]: we like deletion protection for all our databases 😀

File: namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/opensearch.tf [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+#}
+#
+#resource "kubernetes_secret" "os_snapshots_role" {
+#  metadata {
+#    name      = "es-snapshot-role"
+#    namespace = var.namespace
+#  }
+#
+#  data = {
+#    snapshot_role_arn = module.hmpps_prisoner_search_opensearch.snapshot_role_arn
```
Comment (petergphillips) [inline namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-prisoner-search-dev/resources/opensearch.tf:49]: I've had to temporarily comment out this block - only `proxy_url` seems to be exposed by the module so not sure how to do snapshots?

**Review by jakemulley — APPROVED** [review]

**Pull request merged**